src/ray/util/process.cc

```diff
@@ -528,6 +532,55 @@ void Process::Kill() {
   }
 }
 
+namespace {
+/// Tracks subprocesses' exit status.
+class ProcessExitStatusTracker {
+ public:
+  static ProcessExitStatusTracker &GetInstance() {
+    static ProcessExitStatusTracker instance;
+    return instance;
+  }
+
+  void SetExitStatus(pid_t pid, int status) {
+    if (pid <= 0) {
+      return;
+    }
+    absl::WriterMutexLock lock(&mutex_);
+    process_exit_status_[pid] = status;
+  }
+
+  std::string GetExitStatus(pid_t pid) {
+    absl::ReaderMutexLock lock(&mutex_);
+    auto it = process_exit_status_.find(pid);
+    if (it == process_exit_status_.end()) {
+      return "Unknown";
+    }
```

Comment:
Consider adding this link to the comment? https://www.gnu.org/software/libc/manual/html_node/Process-Completion-Status.html